ql/instruments/zerocouponswap.hpp

```diff
@@ -0,0 +1,135 @@
+                           L(T_{k},T_{k+1})) -1 \right],
+        \f]
+        where \f$ L(T_{i}, T_{j})) \f$ are interest rate index fixings
+        for accrual period \f$ [T_{i}, T_{j}] \f$.
+        For simple averaging we have:
+        \f[
+        N^{FLT} = N \left[ \sum_{k=0}^{K} \alpha(T_{k},T_{k+1})
+                           L(T_{k},T_{k+1}) \right].
+        \f]
+        For a par contract, it holds that:
+        \f[
+        P_n(0,T) N^{FIX} = P_n(0,T) N^{FLT}
+        \f]
+        where \f$ T \f$ is the final payment time, \f$ P_n(0,t) \f$
+        is the nominal discount factor at time \f$ t \f$.
+
+        At maturity the two single cashflows are swapped.
+
+        \note we do not need Schedules on the legs because they use
+              one or two dates only per leg. Those dates are not 
+              adjusted for potential non-business days. Only the
+              payment date is subject to adjustment.
+    */
+
+    class ZeroCouponSwap : public Swap {
+      public:
+        enum Type { Receiver = -1, Payer = 1 };
+        
+        ZeroCouponSwap(Type type,
+                       Real baseNominal,
+                       const Date& startDate,
+                       const Date& maturityDate, 
+                       Real fixedPayment,
+                       ext::shared_ptr<IborIndex> iborIndex,
+                       const Calendar& paymentCalendar,
+                       BusinessDayConvention paymentConvention = Following,
+                       Natural paymentDelay = 0,
+                       RateAveraging::Type averagingMethod = RateAveraging::Compound);
+
+        ZeroCouponSwap(Type type,
+                       Real baseNominal,
+                       const Date& startDate,
+                       const Date& maturityDate,
+                       Real fixedRate,
+                       const DayCounter& fixedDayCounter,
+                       ext::shared_ptr<IborIndex> iborIndex,
+                       const Calendar& paymentCalendar,
+                       BusinessDayConvention paymentConvention = Following,
+                       Natural paymentDelay = 0,
+                       RateAveraging::Type averagingMethod = RateAveraging::Compound);
+
+        //! \name Inspectors
+        //@{
+        //! "payer" or "receiver" refer to the fixed leg.
+        Type type() const { return type_; }
+        Real baseNominal() const { return baseNominal_; }
+        Real fixedPayment() const { return fixedPayment_; }
+        const ext::shared_ptr<IborIndex>& iborIndex() const { return iborIndex_; }
+        RateAveraging::Type averagingMethod() const { return averagingMethod_; }
+
+        //! just one cashflow in each leg
+        const Leg& fixedLeg() const;
+        //! just one cashflow in each leg
+        const Leg& floatingLeg() const;
+        //@}
+
+        //! \name Results
+        //@{
+        Real fixedLegNPV() const;
+        Real floatingLegNPV() const;
+        Real fairFixedPayment() const;
```

Comment:
It might be useful to also have a `Rate fairFixedRate(const DayCounter&)`

Reply:
I added such method with a corresponding unit test.